interp/interp_test.go

```
@@ -2519,6 +2571,135 @@ func TestInterpreter_Unmarshal(t *testing.T) {
+
+		var add func(context.Context, int32, int32) (int32, error)
+		require.NoError(t, i.Unmarshal(fn, &add))
+		got, err := add(context.Background(), 2, 3)
+		require.NoError(t, err)
+		require.Equal(t, int32(5), got)
+	})
+
+	t.Run("VM function context identity", func(t *testing.T) {
+		var got context.Context
+		i := New(program.New(nil), WithTick(2), WithHook(func(i *Interpreter) error {
+			got = i.Context()
```

golangci-lint (2.12.2)

[error] nested context in function literal

(fatcontext)